src/audio/module_adapter/module_adapter.c

```diff
@@ -48,7 +48,7 @@ struct comp_dev *module_adapter_new(const struct comp_driver *drv,
 
 	if (!config) {
 		comp_cl_err(drv, "module_adapter_new(), wrong input params! drv = %x config = %x",
-			    (uint32_t)drv, (uint32_t)config);
+			    (uint32_t)(uintptr_t)drv, (uint32_t)(uintptr_t)config);
```

Comment:
Double casts are very rarely needed so they need at least a short comment/justification. I see you mentioned something in the commit message but that's too far away from the code.

Also: why not use `%zu` / `(size_t)` here?

Reply:
one could use `PRIxPTR` to print `uintptr_t` directly

Reply:
Thanks for the suggestions! Decided to go with Marc's suggestion for now. Avoiding the PRI* macros for now because of some interesting points from https://github.com/zephyrproject-rtos/zephyr/issues/46032.